NBitcoin/BIP174/PartiallySignedTransaction.cs

```diff
@@ -439,13 +439,16 @@ public override bool Equals(object obj)
 		}
 		public bool Equals(PSBTInput other) =>
 			IsReferencingSamePrevOut(other) &&
-			Utils.DictEqual(partial_sigs, other.partial_sigs) &&
```

Comment:
I think it could worth to simplify this by comparing the serialized versions. That could be a bit slower the first time but at the same time it could be easier to read and maintain given the amount of code that could be removed.

Reply:
You mean serialize the whole `PSBTInput` ? What do you think the current one which uses predicate to be passed on `DictEqual` ?

Reply:
I think your implementation is okay. I just wanted to put this tradeof on the table: simplicity vs small performance gain. I always prefer simplicity if performance is not something critical. But this is just my opinion.